### ISSUE TITLE
Problems with rails 5.1

### ISSUE THREAD

**Issue by koeckc** [jenseng/hair_trigger#62]
I updated an app to rails 5.1 (ruby 2.3.4) using postgres 9.6.1 on OSX Sierra 
Afterwards all db related tasks result in this error
Without the hairtrigger gem the db interactions work

```
Bundler::GemRequireError: There was an error while trying to load the gem 'hairtrigger'.
Gem Load Error is: No connection pool with 'primary' found.
Backtrace for gem load error is:
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:930:in `retrieve_connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/connection_handling.rb:116:in `retrieve_connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/connection_handling.rb:88:in `connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:832:in `connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:839:in `block in method_missing'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:818:in `block in say_with_time'
/Users/const/.rvm/rubies/ruby-2.3.4/lib/ruby/2.3.0/benchmark.rb:293:in `measure'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:818:in `say_with_time'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:838:in `method_missing'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:600:in `method_missing'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger/migrator.rb:20:in `block in included'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger/migrator.rb:19:in `instance_eval'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger/migrator.rb:19:in `included'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger.rb:224:in `include'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger.rb:224:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `block in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:258:in `load_dependency'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hairtrigger.rb:1:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:91:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:91:in `block (2 levels) in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:86:in `each'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:86:in `block in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:75:in `each'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:75:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler.rb:107:in `require'
/Users/const/develop/defectradar/defectradarserver/config/application.rb:7:in `<top (required)>'
/Users/const/develop/defectradar/defectradarserver/Rakefile:5:in `require'
/Users/const/develop/defectradar/defectradarserver/Rakefile:5:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/rake_module.rb:28:in `load'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/rake_module.rb:28:in `load_rakefile'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:687:in `raw_load_rakefile'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:96:in `block in load_rakefile'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:95:in `load_rakefile'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:79:in `block in run'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/lib/rake/application.rb:77:in `run'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/rake:22:in `load'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/rake:22:in `<main>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/ruby_executable_hooks:15:in `eval'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/ruby_executable_hooks:15:in `<main>'
Bundler Error Backtrace:
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:94:in `rescue in block (2 levels) in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:90:in `block (2 levels) in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:86:in `each'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:86:in `block in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:75:in `each'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:75:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler.rb:107:in `require'
/Users/const/develop/defectradar/defectradarserver/config/application.rb:7:in `<top (required)>'
/Users/const/develop/defectradar/defectradarserver/Rakefile:5:in `require'
/Users/const/develop/defectradar/defectradarserver/Rakefile:5:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/ruby_executable_hooks:15:in `eval'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/ruby_executable_hooks:15:in `<main>'
ActiveRecord::ConnectionNotEstablished: No connection pool with 'primary' found.
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:930:in `retrieve_connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/connection_handling.rb:116:in `retrieve_connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/connection_handling.rb:88:in `connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:832:in `connection'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:839:in `block in method_missing'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:818:in `block in say_with_time'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:818:in `say_with_time'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:838:in `method_missing'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activerecord-5.1.1/lib/active_record/migration.rb:600:in `method_missing'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger/migrator.rb:20:in `block in included'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger/migrator.rb:19:in `instance_eval'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger/migrator.rb:19:in `included'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger.rb:224:in `include'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hair_trigger.rb:224:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `block in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:258:in `load_dependency'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/hairtrigger-0.2.19/lib/hairtrigger.rb:1:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:91:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:91:in `block (2 levels) in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:86:in `each'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:86:in `block in require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:75:in `each'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler/runtime.rb:75:in `require'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/bundler-1.14.6/lib/bundler.rb:107:in `require'
/Users/const/develop/defectradar/defectradarserver/config/application.rb:7:in `<top (required)>'
/Users/const/develop/defectradar/defectradarserver/Rakefile:5:in `require'
/Users/const/develop/defectradar/defectradarserver/Rakefile:5:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/ruby_executable_hooks:15:in `eval'
/Users/const/.rvm/gems/ruby-2.3.4@defectradar/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)
```

**Comment by IbrahimSerhat** [comment]
i got same error in Rails 5.1.1

**Comment by wildtype** [comment]
Got same problem here

**Comment by sethgw** [comment]
Same problem after upgrading to 5.1

**Comment by wilianto** [comment]
mind if someone reviewing this issue, there already a PR but no one is reviewing,

**Comment by iagopiimenta** [comment]
@koeckc This has been fixed in the master. It has not been updated in rubygems yet. Use `gem 'hairtrigger', github: 'jenseng/hair_trigger', branch: 'master'` in your gemfile.

**Comment by jenseng** [comment]
Thanks for the bug reports and PRs, I've merged #64 and will cut a new release in the next couple hours 👍
